ocs_ci/utility/networksplit/main.py

```diff
@@ -0,0 +1,272 @@
+
+    Raises:
+        ValueError: in case invalid ``split_name`` or ``target_dt`` is
+            specified.
+
+    """
+    # input validation
+    if split_name not in VALID_NETWORK_SPLITS:
+        raise ValueError(f"invalid split_name specified: '{split_name}'")
+    now_dt = datetime.now()
+    # TODO: measure and verify 10 minute limit
+    if target_dt - now_dt <= timedelta(minutes=10):
+        msg = (
+            "target start time is not at least 10 minutes in the future, "
+            "and it's not possible to guarantee that start timers will be "
+            "scheduled across all nodes in time"
+        )
+        logger.error(msg)
+        raise ValueError(msg)
+    # convert start timestamp into unix time (number of seconds since epoch)
+    start_ts = int(target_dt.timestamp())
+    # compute target stop timestamp
+    stop_ts = start_ts + (target_length * 60)
+    # generate systemd timer unit names
+    start_unit = f"network-split-{split_name}-setup@.timer@${start_ts}.timer"
+    stop_unit = f"network-split-teardown@${stop_ts}.timer"
+    # schedule both timers on every node of the cluster
+    for node in _list_cluster_nodes():
+        node_name = node[5:]
+        cmd_str = f"systemctl start {start_unit} {stop_unit}"
+        exec_oc_debug_cmd(node_name, cmd_str, timeout=300)
+
+
+def main_setup():
+    """
+    Simple command line interface to generate MachineConfig yaml to deploy to
+    make scheduling network splits possible.
+
+    Example usage::
+
+         $ ocs-network-split-setup --zone-label-names arbiter,d1,d2 -o mc.yaml
+         $ oc create -f mc.yaml
+         $ oc get mcp
+
+    """
+    ap = argparse.ArgumentParser(description="network split setup helper")
+    ap.add_argument(
+        "--print-env-only",
+        action="store_true",
+        default=False,
+        help="just show firewall zone env file and exit",
+    )
+    ap.add_argument(
+        "--output",
+        "-o",
+        metavar="FILE",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="name of yaml file with MachineConfig to deploy on OCP cluster",
+    )
+    ap.add_argument(
+        "--zone-label-names",
+        required=True,
+        help="comma separated list of zone names 'a,b,c' where a is an arbiter",
+    )
+    ap.add_argument(
+        "-d",
+        "--debug",
+        action="store_true",
+        help="set log level to DEBUG",
+    )
+    args = ap.parse_args()
+
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+
+    # get node ip addresses of each zone via zone config
+    zone_names = args.zone_label_names.split(",")
+    zone_config = get_zone_config(zone_names)
+    zone_env = zone_config.get_env_file()
+
+    if args.print_env_only:
+        print(zone_env)
+        return
+
+    # get MachineConfig spec (ready to deploy list of dics)
+    mc = get_networksplit_mc_spec(zone_env)
+    args.output.write(yaml.dump_all(mc))
+
+
+def main_schedule():
```

Comment:
is the intention to use this from a test or just as an outside tool?

Reply:
Both `main_*` functions implements command line tools, the rest of public methods are expected to be used by a fixture or a test.